src/pages/Trace/AddMeasurement/AddMeasurement.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useRef, ChangeEvent } from 'react';
+import './AddMeasurement.css';
+import checkIcon from '../../../assets/check-icon.svg';
+import { renderer, pathDocument } from 'canvas/Renderer';
+import { InstructionModal } from 'components/InstructionModal/InstructionModal';
+import { Input } from 'components/Input/Input';
+import { NavButton } from 'components/NavButton/NavButton';
+import { ActionButton } from 'components/ActionButton/ActionButton';
+
+interface AddMeasurementProps {
+    setUploadedFileData: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const AddMeasurement: React.FC<AddMeasurementProps> = ({ setUploadedFileData }) => {
+    const [inputMeasurement, setInputMeasurement] = React.useState<string>("");
+    
+    // Remove class that puts canvas in the background, remove picture,
+    // and reinitialize event listeners.
+    const canvasContainerRef = useRef(document.getElementsByClassName('canvasContainer')[0]);
+    useEffect(() => {
+        canvasContainerRef.current = document.getElementsByClassName('canvasContainer')[0];
+        canvasContainerRef.current.classList.remove('canvasContainerBackground');
+        canvasContainerRef.current.appendChild(renderer.measurementInit());
+        setUploadedFileData("");
+    }, [canvasContainerRef, setUploadedFileData]);
+
+    const handleSubmit = () => {
+        if (parseFloat(inputMeasurement).toString() !== inputMeasurement.trim()) {
+            // TODO: pop an error modal to the user
+            console.log('Please enter a decimal number');
+        } else {
+            const wasPathSelected = pathDocument.getPatternPaths().some(path => path.isSelected());
+            if (!wasPathSelected) {
+                // TODO: pop an error modal to the user
+                console.log('Please select a path');
+            } else {
+                console.log('ok');
+                pathDocument.setSizeRatio(parseFloat(inputMeasurement));
+            }
+        }
+    };
+
+    const instructModal: InstructModal = {text: ['Choose a path to measure.']};
+
+    const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
handleChange could be renamed  handleInput to be more specific.

Reply:
I disagree. Handlechange is triggered every time the user types a new character in the input, not on input submission. I would agree on something like handleInputChange though. Let me know what you think.

Reply:
Ah, that's right. handleInputChange sounds good to me :)

Reply:
ok, done

Reply:
ok, done!